lib/stdlib/src/json.erl

```diff
@@ -0,0 +1,864 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 2024-2024. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%% % @format
+%%
+-module(json).
+
+-export([
+    encode/1, encode/2,
+    encode_value/2,
+    encode_atom/2,
+    encode_integer/1,
+    encode_float/1,
+    encode_list/2,
+    encode_map/2,
+    encode_map_checked/2,
+    encode_key_value_list/2,
+    encode_key_value_list_checked/2,
+    encode_binary/1,
+    encode_binary_escape_all/1
+]).
+-export_type([encoder/0, simple_encode_value/0]).
+
+-export([
+    decode/1, decode/3
+]).
+-export_type([
+    from_binary_fun/0,
+    array_start_fun/0,
+    array_push_fun/0,
+    array_finish_fun/0,
+    object_start_fun/0,
+    object_push_fun/0,
+    object_finish_fun/0,
+    decoders/0
+]).
+
+-compile(warn_missing_spec).
+
+-compile(
+    {inline, [
+        encode_atom/2,
+        encode_integer/1,
+        encode_float/1,
+        encode_object/1,
+        escape/1,
+        escape_binary/1,
+        escape_all/1,
+        utf8t/0,
+        utf8s/0,
+        utf8s0/0,
+        multibyte_size/1
+    ]}
+).
+
+-include("json.hrl").
+-define(UTF8_ACCEPT, 0).
+-define(UTF8_REJECT, 12).
+
+%%
+%% Encoding implementation
+%%
+
+-type encoder() :: fun((dynamic(), encoder()) -> iodata()).
+
+-type simple_encode_value() ::
```

Comment:
I see how in the future many will want to use this type in their own typespecs as "the JSON type". Might it make sense to try to make the name shorter?

Reply:
That's a good question - I'm going to rename those types to just `encode_value` and `decode_value`